var/spack/repos/builtin/packages/py-jaxlib/package.py

```diff
@@ -17,25 +17,52 @@ class PyJaxlib(PythonPackage, CudaPackage):
     tmp_path = ""
     buildtmp = ""
 
+    version("0.4.3", sha256="2104735dc22be2b105e5517bd5bc6ae97f40e8e9e54928cac1585c6112a3d910")
     version("0.3.22", sha256="680a6f5265ba26d5515617a95ae47244005366f879a5c321782fde60f34e6d0d")
     version("0.1.74", sha256="bbc78c7a4927012dcb1b7cd135c7521f782d7dad516a2401b56d3190f81afe35")
 
-    # see jaxlib/setup.py for dependencies
-    depends_on("python@3.7:", type=("build", "run"))
+    # jaxlib/setup.py
+    depends_on("python@3.8:", when="@0.4:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
-
-    depends_on("py-numpy@1.18:", type=("build", "run"), when="@0.1.74")
-    depends_on("py-numpy@1.20:", type=("build", "run"), when="@0.3.22")
+    depends_on("py-numpy@1.20:", when="@0.3:", type=("build", "run"))
+    depends_on("py-numpy@1.18:", type=("build", "run"))
     depends_on("py-scipy@1.5:", type=("build", "run"))
-    depends_on("py-absl-py", type=("build", "run"))
-    depends_on("py-flatbuffers@1.12:2", type=("build", "run"), when="@0.1.74")
-    # Bazel 5 not yet supported: https://github.com/google/jax/issues/8440
-    depends_on("bazel@4.1.0:4", type=("build"), when="@0.1.74")
-    # Bazel 5 support starts here
-    depends_on("bazel@5.1.1:", type=("build"), when="@0.3.22")
+
+    # .bazelversion
+    depends_on("bazel@5.1.1:", when="@0.3:", type="build")
+    # https://github.com/google/jax/issues/8440
+    depends_on("bazel@4.1:4", when="@0.1", type="build")
+
+    # README.md
+    depends_on("cuda@11.4:", when="@0.4:+cuda")
+    depends_on("cuda@11.1:", when="@0.3+cuda")
+    depends_on("cuda@11.1:11.7.0", when="@0.1+cuda")  # TODO: why the upper bound?
```

Comment:
Any idea where this upper bound came from?

Reply:
it's a headerfile change in cuda 11.7.1 that angers the old bundled tf install under the hood - i can dig out my detailed notes if necessary...

Reply:
If you have a link to an issue that would be great, otherwise I'll add a comment to that affect

Reply:
we've been here before https://github.com/google/jax/issues/12614